### ISSUE TITLE
Parser Does Not Like Method Reference Operator (::)

### ISSUE THREAD

**Issue by msimmon6** [Algomorph/CodeGrader#175]
**Describe the bug**

Parse error in a file, 99% sure it is due to the :: operator. The student, by all accounts, definitely did not know what they were doing. We should still probably handle it.

**Where did the bug occur**

Semester: Summer 2022
Instructor & Course: Pedram's CMSC132
Project name on Submit Server: Exam 2
Student directory id: thou1

Code segment (if known & relevant):

```Java
// Just a portion, only one line is important
Collections.sort(intcollect);
int[] arr = new int[intcollect.size()];
Arrays.setAll(arr, intcollect::get);Keyword: Arrays
return arr;
```

**Additional info**

![image](https://user-images.githubusercontent.com/72671782/173727543-7153fafa-a0ba-4281-a718-ad40121f73d3.png)

Looks like it was expecting an operator, but not this one. It's probably a simple fix of adding :: as a valid operator.

**Comment by msimmon6** [comment]
Also occurs in student's P1

**Comment by msimmon6** [comment]
Edit: It's considered a separator https://docs.oracle.com/javase/specs/jls/se15/html/jls-3.html#jls-3.11

**Comment by msimmon6** [comment]
I think the answer is just including "::" in the list of separators at the bottom of your Java.1.15.pegjs file, but the actual parser file in this looks too intimidating to edit... I will leave it to you to fix.

**Comment by Algomorph** [comment]
No problem, thanks for reporting. I've been aware of this issue for over a year now, just these one-off occurrences tend to be super-rare in 131 and 132 student code. Thanks for creating the issue, maybe it will be useful in the future -- but we'll hold off on the fix for now.

**Comment by msimmon6** [comment]
Yeah -- this person has done it twice now so I hope she does not continue :(
